lgc/util/llpcGfxRegHelper.h

```diff
@@ -0,0 +1,181 @@
+// optimized llvm instruction sequences for these states.
+//
+//    -------------------------------------------------------------------------------
+//   | pValue    | isModified |                       state                          |
+//    -------------------------------------------------------------------------------
+//   | nullptr   | false/true | The value is never being touched, need load value    |
+//   | effective | true       | The value is being modified, need reload value       |
+//   | effective | false      | The value is latest cached, can be accessed directly |
+//    -------------------------------------------------------------------------------
+// Note: the value mentioned above are all in symbolic state and the load/reload are symbolic expressions.
+struct BitsState
+{
+    Value* pValue     = nullptr;
+    bool   isModified = false;
+};
+
+// =====================================================================================================================
+// Helper class for handling graphics registers.
+// Note: 1) Don't use GfxRegHelper directly, please implement your own register helper class, such as SqImgSampRegHelper
+//       2) The ID (enum) used in this class is determined by BitsInfo
+//       3) The count of BisState used in this class is determined by BitsInfo
+// e.g.
+//       ID                              BitsInfo                                 BitsState
+//       {                               {                                        {
+//         FilterMode = 0,                 { 0, 30, 2 }, // FilterMode              {nullptr, false},
+//         xyMagFilter,         <==        { 2, 20, 2 }, // XyMagFilter    ==>      {nullptr, false},
+//         xyMinFilter,                    { 2, 22, 2 }, // XyMinFilter             {nullptr, false},
+//       };                              };                                       };
+class GfxRegHelper : public GfxRegHandlerBase
+{
+protected:
+    GfxRegHelper(Builder* pBuilder, Value* pRegister);
+
+    // Common function for getting the current llvm value for the hardware register
+    Value* GetRegCommon(uint32_t id);
+
+    // Common function for setting the current llvm value for the hardware register
+    void SetRegCommon(uint32_t id, Value* pVal)
+    {
+        SetBits(m_bitsInfo[id], pVal);
+        // It is assumed the register is being modified
+        m_bitsState[id].isModified = 1;
+    }
+
+    // Get combined data from two seperate DWORDs
+    Value* GetRegCombine(uint32_t idLo, uint32_t idHi);
+
+    // Set data into two seperate DWORDs
+    void SetRegCombine(uint32_t idLo, uint32_t idHi, Value* pReg);
+
+    // Get current llvm value state for the hardware register
+    const BitsState* GetBitsState() { return m_bitsState; }
+
+    // Set current llvm value state for the hardware register
+    void SetBitsState(BitsState* bitsState) { m_bitsState = bitsState; }
+
+protected:
+    Value*          m_pInt32One     = nullptr;
```

Comment:
m_pInt32One -> m_pOne = nullptr; // Int32 constant one
Since in this file, there is only int32 one. No need of adding the type prefix "Int32".